### PR TITLE
Solid Materials: Cooler Edition

### PR DESCRIPTION
**Pull request by Lmandude**
Same thing as before, but with the fancy-schmancy temperature stuff bay shoehorned in.

### CODE REVIEW

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 1]
```diff
@@ -490,6 +490,98 @@
 	..()
 	new /obj/item/stack/material/phoron(get_turf(holder.my_atom), created_volume)
 
+/datum/chemical_reaction/uraniumsolidification
+	name = "Solid Uranium"
+	result = null
+	required_reagents = list(/datum/reagent/iron = 5, /datum/reagent/uranium = 20)
+	result_amount = 1
+	minimum_temperature = (-80 CELCIUS) - 100
```
Comment (scrdest) [inline code/modules/reagents/Chemistry-Recipes.dm]: Wait - does this mean that at -190 Celsius, it would be _too cold for it to freeze_?

Comment (Lmandude) [reply code/modules/reagents/Chemistry-Recipes.dm]: Yep. I just used the same values bay threw in for phoron solidification, so blame them for that brilliant piece of logic. Or me for being lazy, either works. Can set them all to 0 if need be.

**Comment by Glloyd** [comment]
@Naxodile Do you want to take a look at this one? You know chem a whole lot better than me

File: code/modules/reagents/Chemistry-Recipes.dm (hunk 1 above)
Comment (Naxodile) [reply code/modules/reagents/Chemistry-Recipes.dm:498]: Just having it blank to default to 0 would be preferable, yeah.

**Review by Naxodile — CHANGES_REQUESTED** [review]
Fix the weird temps and we're good to go. I can make a million suggestions to batch commit if things aren't setup on your end.

File: code/modules/reagents/Chemistry-Recipes.dm (hunk 1 above)
Comment (Naxodile) [inline code/modules/reagents/Chemistry-Recipes.dm:498]:
```suggestion
```

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 2]
```diff
@@ -490,6 +490,98 @@
...
+/datum/chemical_reaction/uraniumsolidification/on_reaction(var/datum/reagents/holder, var/created_volume, var/reaction_flags)
+	..()
+	new /obj/item/stack/material/uranium(get_turf(holder.my_atom), created_volume)
+
+/datum/chemical_reaction/goldsolidification
+	name = "Solid Gold"
+	result = null
+	required_reagents = list(/datum/reagent/iron = 5, /datum/reagent/gold = 20)
+	result_amount = 1
+	minimum_temperature = (-80 CELCIUS) - 100
```
Comment (Naxodile) [inline code/modules/reagents/Chemistry-Recipes.dm:511]:
```suggestion
```

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 3]
```diff
@@ -490,6 +490,98 @@
...
+/datum/chemical_reaction/goldsolidification/on_reaction(var/datum/reagents/holder, var/created_volume, var/reaction_flags)
+	..()
+	new /obj/item/stack/material/gold(get_turf(holder.my_atom), created_volume)
+
+/datum/chemical_reaction/silversolidification
+	name = "Solid Silver"
+	result = null
+	required_reagents = list(/datum/reagent/iron = 5, /datum/reagent/silver = 20)
+	result_amount = 1
+	minimum_temperature = (-80 CELCIUS) - 100
```
Comment (Naxodile) [inline code/modules/reagents/Chemistry-Recipes.dm:524]:
```suggestion
```

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 4]
```diff
@@ -490,6 +490,98 @@
...
+/datum/chemical_reaction/silversolidification/on_reaction(var/datum/reagents/holder, var/created_volume, var/reaction_flags)
+	..()
+	new /obj/item/stack/material/silver(get_turf(holder.my_atom), created_volume)
+
+/datum/chemical_reaction/steelsolidification
+	name = "Solid Steel"
+	result = null
+	required_reagents = list(/datum/reagent/iron = 20, /datum/reagent/carbon = 5)
+	result_amount = 1
+	minimum_temperature = (-80 CELCIUS) - 100
```
Comment (Naxodile) [inline code/modules/reagents/Chemistry-Recipes.dm:537]:
```suggestion
```

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 5]
```diff
@@ -490,6 +490,98 @@
...
+/datum/chemical_reaction/steelsolidification/on_reaction(var/datum/reagents/holder, var/created_volume, var/reaction_flags)
+	..()
+	new /obj/item/stack/material/steel(get_turf(holder.my_atom), created_volume)
+
+/datum/chemical_reaction/mhydrogensolidification
+	name = "Solid Metallic Hydrogen"
+	result = null
+	required_reagents = list(/datum/reagent/iron = 5, /datum/reagent/hydrazine = 20)
+	result_amount = 1
+	minimum_temperature = (-80 CELCIUS) - 100
```
Comment (Naxodile) [inline code/modules/reagents/Chemistry-Recipes.dm:550]:
```suggestion
```

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 6]
```diff
@@ -490,6 +490,98 @@
...
+	..()
+	new /obj/item/stack/material/mhydrogen(get_turf(holder.my_atom), created_volume)
+
+/datum/chemical_reaction/ironsolidification
+	name = "Solid Iron"
+	result = null
+	required_reagents = list(/datum/reagent/iron = 25)
+	inhibitors = list(/datum/reagent/carbon = 5)
+	result_amount = 1
+	minimum_temperature = (-80 CELCIUS) - 100
```
Comment (Naxodile) [inline code/modules/reagents/Chemistry-Recipes.dm:564]:
```suggestion
```

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 7]
```diff
@@ -490,6 +490,98 @@
...
+/datum/chemical_reaction/ironsolidification/on_reaction(var/datum/reagents/holder, var/created_volume, var/reaction_flags)
+	..()
+	new /obj/item/stack/material/iron(get_turf(holder.my_atom), created_volume)
+
+/datum/chemical_reaction/aluminumsolidification
+	name = "Solid Aluminum"
+	result = null
+	required_reagents = list(/datum/reagent/iron = 5, /datum/reagent/aluminum = 20)
+	result_amount = 1
+	minimum_temperature = (-80 CELCIUS) - 100
```
Comment (Naxodile) [inline code/modules/reagents/Chemistry-Recipes.dm:577]:
```suggestion
```

**Review by Naxodile — CHANGES_REQUESTED** [review]

**Review by Naxodile — APPROVED** [review]
Good to merge @Glloyd

**Pull request merged**